src/model.cpp

```diff
@@ -135,28 +149,23 @@ time_type model::run(time_type tfinal, time_type dt) {
         PL();
 
         PE("events","from-spikes");
-        auto events = communicator_.make_event_queues(global_spikes);
+        communicator_.make_event_queues(global_spikes, pending_events_);
         PL();
 
         PE("enqueue");
-        threading::parallel_for::apply(0, communicator_.num_local_cells(),
-            [&](cell_size_type i) {
-                const auto epid = epoch_.id;
-                merge_events(
-                    epoch_.tfinal,
-                    epoch_.tfinal+std::min(t_+t_interval, tfinal),
-                    event_lanes(epid)[i],
-                    events[i],
-                    event_generators_[i],
-                    event_lanes(epid+1)[i]);
-            });
+        const auto t0 = epoch_.tfinal;
+        const auto t1 = std::min(tfinal, t0+t_interval);
```

Comment:
Having these tfinal looking this much alike is an error waiting to happen.
What are the local variables and how they are changed in the different steps is also a little unclear.

Reply:
Personally I don't find it too distracting: one is an argument passed to the run method, the other is a member of `epoch`: so long as each variable is clearly defined, the user can figure out what is going on.

The cause of confusion is that local variables, like `tfinal`, are being accessed inside a lambda, which adds an extra layer/scope between variable definition and variable use. We need to wrap the tasks in lambdas to pass them to the threading back end for execution/schedulting. Althernative approaches, like declaring the lamdas directly inline when passing them to the threading back end, or making them stand alone functions complicate code in other ways... I can't see how to make this much clearer.